### PR TITLE
fix: (platform) add switch component

### PR DESCRIPTION
**Pull request by Betrozov**
#### Please provide a link to the associated issue.
fixes https://github.com/SAP/fundamental-ngx/issues/2867

#### Please provide a brief summary of this pull request.
Created the Platform Switch component

#### Please check whether the PR fulfills the following requirements

- [X] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [X] tests for the changes that have been done
- [X] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [X] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: apps/docs/src/app/platform/component-docs/platform-forms/switch/switch-header/switch-header.component.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<header>Switch</header>
+<description>
+    <p>Switch description</p>
+</description>
+<import module="SwitchModule"></import>
```
Comment (fkolar) [inline apps/docs/src/app/platform/component-docs/platform-forms/switch/switch-header/switch-header.component.html:5]: Missing actual description what we usually add to API docs:

Checkout other components in the platform where we not try to say the same what you can read in core but at minimum how we are enhancing this.

File: libs/platform/src/lib/components/form/switch/switch/switch.config.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import { ContentDensity } from '../../../form/form-control';
+
+/**
+ * Default options for platform Switch
+ */
+@Injectable({ providedIn: 'root' })
+export class SwitchConfig {
+    /**
+     * Content Density of element. 'cozy' | 'compact'
+     */
```
Comment (fkolar) [inline libs/platform/src/lib/components/form/switch/switch/switch.config.ts:12]: We could have more generic config as content density can be applied to all the components like this. I remember Dima was showing me `PlatformConfig`

Comment (valorkin) [reply libs/platform/src/lib/components/form/switch/switch/switch.config.ts:12]: It wasn't yet approved, so let's stick with component level configs, and later we could came up with platform/core/ or even all level configs and services

Comment (fkolar) [reply libs/platform/src/lib/components/form/switch/switch/switch.config.ts:12]: sounds good if its one component but if we need to keep config for each new component like this, I would go in the next iteration to more generic approach and refactor this.

Comment (valorkin) [reply libs/platform/src/lib/components/form/switch/switch/switch.config.ts:12]: components specific configs are must have anyway, over time they will have even more properties

**Comment by JKMarkowski** [comment]
Hi @Betrozov,
I see that whole switch component is implemented from scratch, wasn't it supposed to use implementation from core?

**Comment by fkolar** [comment]
There is missing disabled case in fundamentals- styles

File: libs/platform/src/lib/components/form/switch/switch/switch.component.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    templateUrl: './switch.component.html',
+    styleUrls: ['./switch.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    providers: [{ provide: FormFieldControl, useExisting: SwitchComponent, multi: true }]
+})
+export class SwitchComponent extends BaseInput {
+    /** aria-label attribute of the inner input element. */
+    @Input()
+    ariaLabel = 'Switch input';
```
Comment (JKMarkowski) [inline libs/platform/src/lib/components/form/switch/switch/switch.component.ts:38]: I know it's detail, but could you add type there?

File: libs/platform/src/lib/components/form/switch/switch/switch.component.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+     */
+    @Input()
+    set contentDensity(contentDensity: ContentDensity) {
+        this._contentDensity = contentDensity;
+        this.isCompact = contentDensity === 'compact';
+    }
+
+    /** Whether the switch is semantic */
+    @Input()
+    semantic = false;
```
Comment (JKMarkowski) [inline libs/platform/src/lib/components/form/switch/switch/switch.component.ts:55]: Let's add missing type

File: libs/platform/src/lib/components/form/switch/switch/index.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export * from './switch.component';
```
[the remark this answers is not in the document]
Comment (Betrozov) [reply libs/platform/src/lib/components/form/switch/switch/index.ts:1]: it is for decreasing the path when we import this component.

File: libs/platform/src/lib/components/form/switch/switch/switch.component.ts (hunk 3 above)
Comment (fkolar) [reply libs/platform/src/lib/components/form/switch/switch/switch.component.ts:35]: In this you dont really need to. as you assign a string, this is default behavior for Lint Rules. We dont have general guidelines for this and this is some thing to be noted as well and decide on the direction. I would leave this as it is as this is also proper form.

File: libs/platform/src/lib/components/form/switch/switch/index.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (fkolar) [reply libs/platform/src/lib/components/form/switch/switch/index.ts:1]: Its common practice to use barrel files.

File: apps/docs/src/app/platform/platform-documentation.routes.ts [hunk 6]
```diff
@@ -4,91 +4,97 @@ import { PlatformHomeComponent } from './component-docs/platform-home/platform-h
...
-            {
-                path: 'textarea',
-                loadChildren: () =>
-                    import('./component-docs/platform-forms/platform-textarea/platform-textarea-docs.module').then(
-                        (m) => m.PlatformTextareaDocsModule
-                    )
-            }
-        ]
-    }
+  {
```
[the remark this answers is not in the document]
Comment (Betrozov) [reply apps/docs/src/app/platform/platform-documentation.routes.ts]: fixed

**Comment by Betrozov** [comment]
> The styling for disabled looks not proper
> in platform
> Screen Shot 2020-07-28 at 7 33 32 AM
> 
> in core
> Screen Shot 2020-07-28 at 7 31 57 AM

fixed

File: apps/docs/src/app/platform/component-docs/platform-forms/switch/switch-header/switch-header.component.html (hunk 1 above)
Comment (Betrozov) [reply apps/docs/src/app/platform/component-docs/platform-forms/switch/switch-header/switch-header.component.html:8]: fixed

**Comment by Betrozov** [comment]
> I added some comments regarding code.
> There is also different focus ring on switch in platform.
> Core:
> image
> 
> Platform:
> image

fixed

**Comment by I074914** [comment]
Behaviour of switch toggle is not working as expected. When we click on the "Default Semantic Compact Switch" and "Default Semantic Compact Switch" on first mouse click it go highlighted. and second click it will work. 
Expected: On first mouse click toggle should work.

**Comment by I074914** [comment]
> > Behaviour of switch toggle is not working as expected. When we click on the "Default Semantic Compact Switch" and "Default Semantic Compact Switch" on first click it is highlighted and second click it will work. But the issue is on first click switch toggle is not working.
> > Expected: On first mouse click toggle should work.
> 
> Fixed

Tested it is working fine. Thanks

**Comment by manjunathanagaraj** [comment]
Looks fine to me. Accessbility testing will be verified by Purushotham.

**Pull request merged**